src/components/AzFileUpload.vue

```diff
@@ -0,0 +1,132 @@
+            <div class="az-drop-file"
+                 :style="{'height': height}"
+                 @drop.prevent="onDropFiles($event.dataTransfer.items)"
+                 @dragover.prevent="">
+                <input id="azFileSelector" type="file" multiple class="input-file"
+                       :name="uploadFieldName"
+                       :accept="accept"
+                       @change="onSelectFiles($event.target.files)"/>
+                <div class="center">
+                    <slot></slot>
+                </div>
+            </div>
+        </form>
+        <az-file-progress></az-file-progress>
+    </div>
+</template>
+
+<script>
+
+    import AzFileProgress from './AzFileProgress'
+
+    export default {
+        props: {
+            accept: {
+                type: String,
+                default: '*'
+            },
+            repository: {
+                type: String,
+                required: true
+            },
+            thumbnail: {
+                type: Boolean,
+                default: false
+            },
+            height: {
+                type: String,
+                default: '200px'
+            }
+        },
+        data() {
+            return {
+                uploadFieldName: 'file'
+            }
+        },
+        methods: {
+            onSelectFiles(fileList) {
+                if (!fileList.length) {
+                    return;
+                }
+                this.uploadFiles(fileList)
+            },
+            onDropFiles(eventItems) {
+                if (!Object.keys(eventItems).length) {
+                    return;
+                }
+                const files = []
+                Object.keys(eventItems).forEach((key) => {
+                    const item = eventItems[key]
+                    if (item.kind === 'file') {
+                        files.push(item.getAsFile())
+                    }
+                })
+                this.uploadFiles(files)
+            },
+            uploadFiles(fileList) {
+                Array
+                    .from(Array(fileList.length).keys())
+                    .map(x => {
+                        const payload = this.createPayload(fileList[x])
+                        this.$store.dispatch('uploadFile', payload)
+                    });
+            },
+            createPayload(file) {
+                return {
+                    repository: this.repository,
+                    thumbnail: this.thumbnail,
+                    formData: this.createFormData(file),
+                    filename: file.name
+                }
+            },
+            createFormData(file) {
+                const formData = new FormData()
+                formData.append(this.uploadFieldName, file)
+                return formData
+            },
+            openFileSelector() {
+                document.getElementById('azFileSelector').click()
+            }
+        },
+        components: {
```

Comment:
Mova os components para o início do arquivo para seguirmos um padrão de organização do Vuejs